azurerm/_modules/aks/log_analytics.tf

```diff
@@ -1,4 +1,5 @@
 resource "azurerm_log_analytics_workspace" "current" {
+  count               = var.enable_log_analytics == true ? 1 : 0
```

Comment:
Adding the count here will require a manual task `terraform state mv` during upgrades. This isn't a big issue, but something I like to avoid as much as possible.

Are the `azurerm_log_analytics_workspace` and the `azurerm_log_analytics_solution` the resources that incur charges?

Reply:
It's the `azurerm_log_analytics_solution` that turns out to be expensive. As I assume workspaces are basically free and of rather little consequence if nothing is stored/deployed in them, I don't really have any issues leaving it, empty. 

Not as elegant, but not strictly wrong

Reply:
Hm, actually I'm not entirely certain by that, as the pricing tier is associated with the workspace, not a particular solution. Nevertheless, the default (and as of right now not configurable) `sku` is priced per GB.

In the "disabled" case could also use the [`Free`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/log_analytics_workspace) for the `sku`.